lib/internal/validators.js

```diff
@@ -257,6 +257,24 @@ const validateObject = hideStackFrames(
     }
   });
 
+/**
+ * @callback validateDictionary - We are using the Web IDL Standard definition
+ *                                of "dictionary" here, which means any value
+ *                                whose Type is either Undefined, Null, or
+ *                                Object (which includes functions).
+ * @param {*} value
+ * @param {string} name
+ * @see https://webidl.spec.whatwg.org/#es-dictionary
```

Comment:
I'd suggest to include https://tc39.es/ecma262/#table-typeof-operator-results to explain why functions are included - typeof returns "functions" for objects that implement [[Call]]

```suggestion
 * @see https://webidl.spec.whatwg.org/#es-dictionary
 * @see https://tc39.es/ecma262/#table-typeof-operator-results
```